### ISSUE TITLE
Making logging optional while keeping results from stdout

### ISSUE THREAD

**Issue by alping** [PrefectHQ/prefect-shell#67]
# Expectation / Proposal

When using ShellOperation there is no way to get the result from the stdout without also logging it (at INFO logging level). This is causing issues when the output to stdout is long as it clutters up the logs and it also seems to slow down the tasks considerably as well as crashing the logger if there are too many tasks doing this simultaneously. 

`Crash detected! Execution was cancelled by the runtime environment.`

Setting `stream_output=False` removes the logging, but also sets the stdout and stderr to DEVNULL, so no result is returned.

https://github.com/PrefectHQ/prefect-shell/blob/4ecab9ed9b038b17a05e15c5de0f1aed22ac9d84/prefect_shell/commands.py#L159-L160

https://github.com/PrefectHQ/prefect-shell/blob/4ecab9ed9b038b17a05e15c5de0f1aed22ac9d84/prefect_shell/commands.py#L302-L309

It would be great if we could separate these into two options, e.g. `return_output` and `log_output`.

# Example

I'm trying to extract information from a set of PDF documents.
The first step is to convert the PDF to text using the [pdftotext](http://www.xpdfreader.com/) command-line utility.
To limit the reading and writing to intermediate files, I'm getting the resulting text back from stdout.

In the terminal this works well:
`pdftotext -raw -enc UTF-8 -nopgbrk -margint 50 -marginb 50 -f 2 "/path/to/file.pdf" -`

When trying to do this with a mapped Prefect task, it works for a few PDF files (but clutters the logs), but runs very slow and eventually crashes when mapping over a bigger list of files.

```python
from pathlib import Path

from prefect import flow, task
from prefect_shell import ShellOperation


@task()
def pdf_to_txt(pdf_file: Path) -> list[str]:
    """Convert a PDF file text."""

    with ShellOperation(
        commands=[
            f'pdftotext -raw -enc UTF-8 -nopgbrk -margint 50 -marginb 50 -f 2 "{pdf_file}" -'
        ]
    ) as pdftotext_operation:
        pdftotext_process = pdftotext_operation.trigger()
        pdftotext_process.wait_for_completion()
        txt_content = pdftotext_process.fetch_result()

    return txt_content


@flow
def extract_pdf_files(pdf_files):
    txt_content = pdf_to_txt.map(pdf_files)

    # ...do something with the text content


if __name__ == "__main__":
    pdf_files = list(Path("documents").glob("*.pdf"))
    extract_pdf_files(pdf_files)
```

- [x] I would like to [help contribute](https://PrefectHQ.github.io/prefect-shell/#contributing) a pull request to resolve this! (But I'm not sure how)

**Comment by ahuang11** [comment]
Thanks for pointing this out and volunteering to help contribute a PR!

[This guide](https://prefecthq.github.io/prefect-shell/#contributing) should help you get started on contributing.

In terms of implementation, I believe we can remove the in-line if else statement, dropping the DEVNULL, and only keeping subprocess.PIPE [here](https://github.com/PrefectHQ/prefect-shell/blob/main/prefect_shell/commands.py#L304-L305). Then, I think [this](https://github.com/PrefectHQ/prefect-shell/blob/main/prefect_shell/commands.py#L159-L160) should prevent it from logging, and it would simply output the stdout/stderr.

Let me know if you need further guidance!

**Comment by alping** [comment]
This would work well for what I want, but is there a reason for the current implementation? I.e. is there any situation where it is beneficial to make sure that the stdout and stderr go to DEVNULL (e.g. performance)?

I'll try to make a PR based on your instructions.

**Comment by ahuang11** [comment]
Thank you for helping contribute; hoping to get this released sometime later today after closing #70.

**Comment by ahuang11** [comment]
Released!
